recaptchaenterprise/samples/snippets/create_mfa_assessment.py

```diff
@@ -0,0 +1,101 @@
+    project_id: str, recaptcha_site_key: str, token: str, recaptcha_action: str, hashed_account_id: str,
+        email: str, phone_number: str
+) -> None:
+    """  Creates an assessment to obtain Multi-Factor Authentication result.
+         If the result is unspecified, sends the request token to the caller to initiate MFA challenge.
+    Args:
+        project_id: GCloud Project ID
+        recaptcha_site_key: Site key obtained by registering a domain/app to use recaptcha services.
+        token: The token obtained from the client on passing the recaptchaSiteKey.
+               To get the token, integrate the recaptchaSiteKey with frontend. See,
+               https://cloud.google.com/recaptcha-enterprise/docs/instrument-web-pages#frontend_integration_score
+        recaptcha_action: The action name corresponding to the token.
+        hashed_account_id: Create hashedAccountId from user identifier.
+                           It's a one-way hash of the user identifier: HMAC SHA 256 + salt
+        email: Email id of the user to trigger the MFA challenge.
+        phone_number: Phone number of the user to trigger the MFA challenge.
+    """
+
+    client = recaptchaenterprise_v1.RecaptchaEnterpriseServiceClient()
+
+    # Set the properties of the event to be tracked.
+    event = recaptchaenterprise_v1.Event()
+    event.site_key = recaptcha_site_key
+    event.token = token
+    event.hashed_account_id = hashed_account_id
+
+    # Set the email address and the phone number to trigger/ verify the MFA challenge.
+    endpoint_verification_info = recaptchaenterprise_v1.EndpointVerificationInfo()
+    endpoint_verification_info.email_address = email
+    endpoint_verification_info.phone_number = phone_number
+
+    account_verification_info = recaptchaenterprise_v1.AccountVerificationInfo()
+    account_verification_info.endpoints = endpoint_verification_info
+
+    assessment = recaptchaenterprise_v1.Assessment()
+    assessment.event = event
+    assessment.account_verification = account_verification_info
+
+    project_name = f"projects/{project_id}"
+
+    # Build the assessment request.
+    request = recaptchaenterprise_v1.CreateAssessmentRequest()
+    request.assessment = assessment
+    request.parent = project_name
+
+    # Check integrity of the response.
+    response = client.create_assessment(request)
+    if not verify_response_integrity(response, recaptcha_action):
+        raise Exception("Failed to verify token integrity.")
+
+    result = response.account_verification.latest_verification_result
+    # If the result is unspecified, send the request token to trigger MFA in the client.
+    # You can choose to send both the email and phone number's request token.
+    if result == recaptchaenterprise_v1.types.AccountVerificationInfo.Result.RESULT_UNSPECIFIED:
+        # Send the request token for assessment. The token is valid for 15 minutes.
+        print("Result unspecified. Triggering MFA challenge.")
```

Comment:
Where do we trigger this challenge? I don't see any specific value returned, or method called inside this `if` statement. All it does is print a message - is this intentional?

Reply:
This challenge will be triggered in the client. I've updated the comment for more clarity. PTAL